draft-ietf-mls-architecture.md

```diff
@@ -440,23 +441,29 @@ latter delivery pattern via unicast channels (sometimes known as "client
 fanout"), broadcast channels ("server fanout"), or a mix of both.
 
 For the most part, MLS does not require the Delivery Service to deliver messages
-in any particular order. The one exception is that, because an MLS group has a
-linear history, the members of the group must agree on the order in which
+in any particular order. Applications can set policies that control their
+tolerance for out-of-order messages (see {{operational-requirements}}), and
+messages that arrive significantly out-of-order will be dropped without
+otherwise affecting the protocol. There are two exceptions to this. First,
+Proposal messages must all arrive before the Commit that references them.
+Second, because an MLS group has a linear history of epochs, the members of the
```

Comment:
How about the requirement that a Commit creating an epoch arrives before any other messages which use that epoch?
[If that is not a strict requirement then we need to discuss how long a client will "hold" an application message for a newer epoch. There are potential DoS issues holding it too long.]

Reply:
Added a note in Operational Requirements that messages may need to be buffered